src/game/server/gamecontext.cpp

```diff
@@ -4478,3 +4496,36 @@ bool CGameContext::IsSpectatorCID(int ClientID) {
 	auto& vec = Server()->spectators_id;
 	return std::find(vec.begin(), vec.end(), ClientID) != vec.end();
 }
+
+void CGameContext::Converse(int ClientID, char *pStr)
+{
+	CPlayer *pPlayer = m_apPlayers[ClientID];
+	if (!pPlayer)
+		return;
+	if(Server()->GetClientSession(ClientID) && Server()->GetClientSession(ClientID)->m_MuteTick > 0)
+	{
+		int Time = Server()->GetClientSession(ClientID)->m_MuteTick/Server()->TickSpeed();
+		SendChatTarget_Localization(ClientID, CHATCATEGORY_ACCUSATION, _("You are muted for {sec:Duration}"), "Duration", &Time, NULL);
+		return;
+	}
+
+	for(int i = 0; i < MAX_CLIENTS; i++)
```

Comment:
i don't understand why the loop is required here and why `m_apPlayers[i]` is checked. Could you please explain? Maybe you could add a comment here that would explain what is going on here?

Reply:
Oh, this is my inattention, and really strange, I'll fix it now